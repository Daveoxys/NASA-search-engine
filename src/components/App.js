import React, { useState } from "react";
import "../styles/App.css";
import Search from "./Search";
import SearchResults from "./searchResults";

function App() {
  const [searchResults, setSearchResults] = useState([]);
  return (
    <div className="app">
      <img
        className="nasa-logo"
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="nasaLogo"
      />
      <Search setSearchResults={setSearchResults} />
      <SearchResults results={searchResults} />
    </div>
  );
}

export default App;
