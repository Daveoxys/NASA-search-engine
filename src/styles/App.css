.app {
  display: flex;
  flex-direction: column;
  text-align: center;
  border-color: blue;
}

.nasa-logo {
  margin: auto;
  width: 500px;
  height: 300px;
}

@media screen and (min-width: 1024px) {
  .nasa-logo {
    width: 30%;
  }
}
