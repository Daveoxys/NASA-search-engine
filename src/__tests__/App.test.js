import React, { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  const { asFragment } = render(<App />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  test("It renders logo", () => {
    render(<App />);
    const nasaLogo = screen.getByAltText("nasaLogo");

    expect(nasaLogo).toBeInTheDocument();
    expect(nasaLogo).toHaveClass("nasa-logo");
  });
});
