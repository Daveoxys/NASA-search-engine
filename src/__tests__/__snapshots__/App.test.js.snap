// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="nasaLogo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <div
      class="Search"
    >
      <form
        class="search-form"
      >
        <input
          class="search-input"
          type="text"
        />
         
        <br />
         
        <br />
        <button
          class="search-b"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <br />
    <br />
    <p>
      We've searched the entirity of an ever expanding universe and found 0 results!
    </p>
  </div>
</DocumentFragment>
`;
